backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,384 @@
+import { ISequence } from '../../model/sequence/ISequence';
+import { ISlide } from '../../model/slide/ISlide';
+import { ISequenceWithSlides } from '../../model/sequence/ISequenceWithSlides';
+import { DBSequence } from '../../model/sequence/DBSequence';
+import { DBSlide } from '../../model/slide/DBSlide';
+import { LoernwerkError, LoernwerkErrorCodes } from '../loernwerkUtilities';
+import { DBUser } from '../../model/user/DBUser';
+/**
+ * Manages the sequence data in the database and handles inquiries requests regarding these
+ */
+export class SequenceController {
+    /**
+     * Creates a new sequence in the database with the given title and given user as owner.
+     * @param name the title of the sequence
+     * @param userId the userid of the owner
+     * @returns the created sequence
+     */
+    public static async createNewSequence(
+        name: string,
+        userId: number
+    ): Promise<ISequence> {
+        if ((await DBUser.findBy({ id: userId })).length !== 1) {
+            throw new LoernwerkError(
+                'user not found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const seq = new DBSequence();
+        seq.creationDate = new Date();
+        let gencode = this.genCode();
+        while ((await DBSequence.findBy({ code: gencode })).length > 0) {
+            gencode = this.genCode();
+        }
+        seq.code = gencode;
+        seq.authorId = userId;
+        seq.name = name;
+        seq.modificationDate = new Date();
+        seq.slideCount = 0;
+        seq.writeAccess = [];
+        seq.readAccess = [];
+        seq.save();
+        return seq;
+    }
+
+    /**
+     * Searches the database for a sequence with corresponding access code. Throws an error if no sequence could be found.
+     * @param code the code of the sequence
+     * @returns the sequence
+     */
+    public static async getSequenceByCode(code: string): Promise<ISequence> {
+        const seq = await DBSequence.findBy({ code: code });
+        if (seq.length === 0) {
+            throw new LoernwerkError(
+                'no matching sequence',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        if (seq.length > 1) {
+            throw new LoernwerkError(
```

Comment:
This cant ever happen - code is a primary key, so our DB should already prevent duplicate keys - therefore, we can also use `findOneBy` above